include/klee/Expr/Expr.h

```diff
@@ -449,33 +450,33 @@ class NotOptimizedExpr : public NonConstantExpr {
 
 /// Class representing a byte update of an array.
 class UpdateNode {
-  friend class UpdateList;  
+  friend class UpdateList;
 
-  mutable unsigned refCount;
   // cache instead of recalc
   unsigned hashValue;
 
 public:
-  const UpdateNode *next;
+  const ref<UpdateNode> next;
   ref<Expr> index, value;
-  
+
+  /// @brief Required by klee::ref-managed objects
+  mutable class ReferenceCounter _refCount;
```

Comment:
This is true in the old version too, but why is this mutable and the Expr one isn't? I assume we want ref manged const updateLists, but not Exprs for some reason? (Thinking about this for #1170 )

Reply:
You just want to refcount a `const` expression tree. Most often, you don't want to change the expressions at all but you want to share it. Therefore, expr const and only counting itself is non-const.